### PR TITLE
Tutorial: Add cMEM estimates in tutorials

### PR DESCRIPTION
**Pull request by Edouard2laire**
Hello, 

As we will be merging https://github.com/multifunkim/best-brainstorm/pull/13 soon, this  PR update the script corresponding to cMEM of the two following tutorials: 

- [x] Add cMEM estimate to `tutorial_epilepsy.m` and add it to tutorial page:
  https://neuroimage.usc.edu/brainstorm/Tutorials/Epilepsy 

- [x] Create `tutorial_BEst.m` which reproduces the steps in the updated Brain Entropy tutorial page:  
  https://neuroimage.usc.edu/brainstorm/Tutorials/TutBEst
  This script the Introduction protocol computed with `tutorial_introduction`

### CODE REVIEW

File: toolbox/script/tutorial_epilepsy.m [hunk 1]
```diff
@@ -282,6 +283,56 @@ function tutorial_epilepsy(tutorial_dir, reports_dir)
...
 
+% Process: Compute sources: BEst
+sAvgSrcMEM = bst_process('CallProcess', 'process_inverse_mem', sFilesAvg, [], ...
+    'comment',     'MEM', ...
+    'mem',         struct(...
+         'MEMpaneloptions', struct(...
+               'InverseMethod', 'MEM', ...
+               'automatic',     struct(...
+                     'MEMexpert',   1, ...
+                     'version',     '3.0.0', ...
```
Comment (rcassani) [inline toolbox/script/tutorial_epilepsy.m:294]: Version `3.0.0`?
This is not available as plugin yet

Comment (Edouard2laire) [reply toolbox/script/tutorial_epilepsy.m:294]: i should merge this PR: https://github.com/multifunkim/best-brainstorm/pull/13 this week moving the version to 3.0.0

Comment (rcassani) [reply toolbox/script/tutorial_epilepsy.m:294]: Perfect, please change this PR to Draft, and back to Read-to-review when its ready

**Comment by Edouard2laire** [comment]
> When trying to run the pipeline, running:
> 
> `sAvgSrcMEM = bst_process('CallProcess', 'process_inverse_mem', sFilesAvg, [], ...`
> 
> gives the error:
> 
> ```
> Line 16: fminusub
> Objective function is undefined at initial point. Fminunc cannot continue.
> _______________________________________________
> Call stack: 
> >fminusub.m at 16 
> >fminunc.m at 488 
> >be_minimize_free_energy.m at 56 
> >be_solve_mem.m at 56 
> >be_launch_mem.m>MEM_mainLoop at 185 
> >be_launch_mem.m at 79 
> >be_cmem_solver.m at 182 
> >be_main_call.m at 157 
> >be_main.m at 60 
> >process_inverse.m>Compute at 696 
> >process_inverse.m at 24 
> >process_inverse_mem.m>Run at 139 
> >process_inverse_mem.m at 24 
> >bst_process.m>Run at 236 
> >bst_process.m>CallProcess at 2316
> >bst_process.m at 38
> ```
> 


Did you download the new MEM version https://github.com/Edouard2laire/best-brainstorm/archive/refs/heads/wMEM.zip? 

this error is due to the fact that the script is having an error with the forward model
image

So for some vertex, G is 0 for all the sensors and it cause issue with MEM. 


> > How can i update the MEM figure from the tutorial ? i would like to change it to
> 
> Firs,t login in the in the Brainstorm website, button at upper right corner. (If you do not have an account, please send me an email, so I can give you the necessary info details on creating one)
> 
> After login, on the tutorial page to edit, go to the top navigatin bar, then Menu > Attachments. Then update the tutorial, use this use this syntax:
> 
> {{attachment:sim_ar_spectra_metrics.png||width="100%"}} More info on working with images on Moinmoin here: http://moinmo.in/HelpOnImages

Thx. :)

**Comment by Edouard2laire** [comment]
also @rcassani, what do you think of the idea to move MEM just after sLoreta ? i think it make more sense to do all the Source analysis: Surface at the same time instead of sLoreta, then volume then back to surface with MEM

**Comment by rcassani** [comment]
I tried with the `master` branch on https://github.com/Edouard2laire/best-brainstorm which is the one used in Brainstorm when installing and updating the `brainentropy` plugin

About the order, I think we can move MEM after sLORETA. With its own section with the [Advance] label.

**Comment by Edouard2laire** [comment]
>> About the order, I think we can move MEM after sLORETA. With its own section with the [Advance] label.


ok. if I may, the tutorial editor is probably the worst editor i had to use to write text so far.

**Comment by rcassani** [comment]
Indeed, writing in MoinMoin is a real pain, apologies for all the suffering.
Few pointers:

* Admin users can revert pages (let me know if you need to do so)
* The slowness things is not on MoinMoin side, but today's issue with the server
* It helps to edit the text without their fancy editor, but using the "Show Raw Text" editor

**Comment by Edouard2laire** [comment]
Hi @rcassani 

it seems that the change in the BEM surface https://github.com/brainstorm-tools/brainstorm3/pull/721/commits/fcce1d7d10135be1952dc76af09c5d7d40227524 to fix the warning about dipole outside the skull is making quite a difference in the inverse problem solution; including for sLoreta: 

Screenshot 2024-07-15 at 21 01 26

Do you know if that is correct ?

**Comment by Edouard2laire** [comment]
> Indeed, writing in MoinMoin is a real pain, apologies for all the suffering. Few pointers:
> 
> * Admin users can revert pages (let me know if you need to do so)
> * The slowness things is not on MoinMoin side, but today's issue with the server
> * It helps to edit the text without their fancy editor, but using the "Show Raw Text" editor

Indeed, the raw text editor is much better ! :)

File: toolbox/script/tutorial_epilepsy.m [hunk 2]
```diff
@@ -224,9 +224,9 @@ function tutorial_epilepsy(tutorial_dir, reports_dir)
 % Process: Generate BEM surfaces
 bst_process('CallProcess', 'process_generate_bem', [], [], ...
     'subjectname', SubjectName, ...
-    'nscalp',      642, ...
-    'nouter',      482, ...
-    'ninner',      482, ...
+    'nscalp',      1922, ...
+    'nouter',      1922, ...
+    'ninner',      1922, ...
```
Comment (rcassani) [inline toolbox/script/tutorial_epilepsy.m:229]: This is not a bug, it was decided so the script can run faster and on computers with limited resources

Comment (Edouard2laire) [reply toolbox/script/tutorial_epilepsy.m]: Done.

**Comment by Edouard2laire** [comment]
https://github.com/multifunkim/best-brainstorm/pull/13 is now merged.  The PR is now ready to be reviewed :)

File: toolbox/script/tutorial_BEst.m [hunk 3]
```diff
@@ -0,0 +1,631 @@
...
+
+%% ===== TUTORIAL #21: NOISE COVARIANCE ==============================================
+%  ===================================================================================
+disp([10 'DEMO> Tutorial #21: Noise covariance' 10]);
+% Process: Compute covariance (noise or data)
+bst_process('CallProcess', 'process_noisecov', sFilesAvgDeviant01, [], ...
+    'baseline',       [], ...
+    'sensortypes',    'MEG', ...
+    'target',         1, ...  % Noise covariance     (covariance over baseline time window)
+    'dcoffset',       0, ...  % Block by block, to avoid effects of slow shifts in data
```
Comment (rcassani) [inline toolbox/script/tutorial_BEst.m:460]: As identity is required, the `dcoffset` is ignored. However, `dcoffset` is a `radio` option, it's value must be `>=1`.

Set it to `1` so it matches the comment of the same line

File: toolbox/script/tutorial_BEst.m [hunk 4]
```diff
@@ -0,0 +1,631 @@
...
+
+% Process: Snapshot: Sources (one time)
+bst_process('CallProcess', 'process_snapshot', sAvgSrwMEM_scaleAll, [], ...
+    'target',    8, ...  % Sources (one time)
+    'modality',  1, ...  % MEG (All)
+    'orient',    1, ...  % left
+    'time',      83.3*1e-3, ...
+    'threshold', 0, ...
+    'Comment',   'Average Devient (wMEM - all scale)');
+%% ===== SAVE REPORT =====
```
Comment (rcassani) [inline toolbox/script/tutorial_BEst.m:622]: Empty line is needed

**Review by rcassani — CHANGES_REQUESTED** [review]
@Edouard2laire, to avoid duplication of code, and potential problems in maintenance.
I think for the `tutorial_BEst.m` we can use the same approach as in the brain fingerprint tutorial script, where it starts with an already processed Protocol.

* https://neuroimage.usc.edu/brainstorm/Tutorials/BrainFingerprint
* https://github.com/brainstorm-tools/brainstorm3/blob/6c646d2248c877a481dcf1e7308ff5287992a676/toolbox/script/tutorial_brain_fingerprint.m#L63-L68

So `tutorial_BEst.m` will look for the the 'TutorialIntroduction' and do the steps starting with L411:

```
%% ===== TUTORIAL #20: HEAD MODEL ====================================================
%  ===================================================================================
```

File: toolbox/script/tutorial_BEst.m [hunk 5]
```diff
@@ -0,0 +1,631 @@
...
+% Process: Snapshot: Sources (one time)
+bst_process('CallProcess', 'process_snapshot', sAvgSrwMEM_scale5, [], ...
+    'target',    8, ...  % Sources (one time)
+    'modality',  1, ...  % MEG (All)
+    'orient',    1, ...  % left
+    'time',      83.3*1e-3, ...
+    'threshold', 0, ...
+    'Comment',   'Average Devient (wMEM - scale 5 )');
+
+% 3. Localizing all scale:
```
[the remark this answers is not in the document]
Comment (rcassani) [reply toolbox/script/tutorial_BEst.m]: "Devient"

**Pull request merged**